### PR TITLE
Cleanup disk circle warnings

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description
This introduced many deprecation warnings to appear all over:
https://github.com/scikit-image/scikit-image/pull/4428

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/draw/_random_shapes.py [hunk 1]
```diff
@@ -166,7 +172,7 @@ def _generate_triangle_mask(point, image, shape, random):
 # Allows lookup by key as well as random selection.
 SHAPE_GENERATORS = dict(
     rectangle=_generate_rectangle_mask,
-    circle=_generate_circle_mask,
+    circle=_generate_disk_mask,
```
Comment (sciunto) [inline skimage/draw/_random_shapes.py]:
```suggestion
    disk=_generate_disk_mask,
```
I suggest to rename this as well

File: skimage/segmentation/morphsnakes.py [hunk 2]
```diff
@@ -101,11 +101,14 @@ def _init_level_set(init_level_set, image_shape):
...
             res = checkerboard_level_set(image_shape)
+        # TODO: remove me in 0.19.0
         elif init_level_set == 'circle':
             res = circle_level_set(image_shape)
+        elif init_level_set == 'disk':
+            res = disk_level_set(image_shape)
         else:
             raise ValueError("`init_level_set` not in "
-                             "['checkerboard', 'circle']")
+                             "['checkerboard', 'circle', 'disk']")
```
Comment (sciunto) [inline skimage/segmentation/morphsnakes.py]: Add a comment : TODO: remove 'circle' in 0.19.0?

File: skimage/draw/_random_shapes.py [hunk 3]
```diff
@@ -103,11 +103,17 @@ def _generate_circle_mask(point, image, shape, random):
     if available_radius < min_radius:
         raise ArithmeticError('cannot fit shape to image')
     radius = random.randint(min_radius, available_radius + 1)
-    circle = draw_circle(point[0], point[1], radius)
+    # TODO: think about how to deprecate this
+    # Unfortunately, renaming `circle` to `disk` directly would break
+    # downstream code that switches between `circle` and disk.
```
Comment (jni) [inline skimage/draw/_random_shapes.py]: This comment isn't super clear. I think it's easier to say just "switching to 'disk' here would be a breaking change for downstream libraries".

I'm also not positive about this PR in general... "square" and "triangle" are technically not filled either...

Comment (hmaarrfk) [reply skimage/draw/_random_shapes.py]: I can restrict this PR to the internal warnings, as opposed to an API change. I would prefer that anyway ;)

**Comment by hmaarrfk** [comment]
See #4467

**Pull request closed without merging**